crates/sui-types/src/object.rs

```diff
@@ -457,6 +454,7 @@ impl Object {
     ) -> Result<Self, ExecutionError> {
         Ok(Object {
             data: Data::Package(MovePackage::from_module_iter(
+                OBJECT_START_VERSION,
```

Comment:
Now that package version is mutable, should it still start from OBJECT_START_VERSION, or should it be the same as everything else based on lamport version?
Related, I wonder if OBJECT_START_VERSION is still used anywhere else? It not, at least we rename it to PACKAGE_START_VERSION if we don't want to apply lamport version to it yet

Reply:
I decided to keep `OBJECT_START_VERSION` because it is still used in a couple of places, for genesis and test objects.

I don't think packages should use lamport versions.  The reason to use lamport versions is related to interactions between child objects and wrapping/unwrapping, which packages don't participate in, so it's more valuable to retain a human comprehensible version for packages (i.e. if you see that the package is at version X, you know it has X antecedent versions).

To support that, I'll need to parametrise this field, so it isn't always `OBJECT_START_VERSION`, but I'll do that in a follow-up PR.  It's not clear what the best way to do that is without seeing how it will be used.